quaireaux/searching/src/binary_search.cairo

```diff
@@ -0,0 +1,34 @@
+use array::SpanTrait;
+
+fn binary_search<T,
+impl TCopy: Copy<T>,
+impl TDrop: Drop<T>,
+impl TEq: PartialEq<T>,
+impl TOr: PartialOrd<T>>(
+    span: Span<T>, val: T
+) -> Option<u32> {
+    // Initial check
+    if (span.len() == 0) {
+        return Option::None(());
+    }
+    let middle = span.len() / 2;
+    if (*span[middle] == val) {
```

Comment:
In if statements You don't need parenthesis (unless you like them, then it is fine and you can ignore this comment)

Reply:
It was intended like that, I prefer the parenthesis :)